### ISSUE TITLE
does keras support array broadcasting?

### ISSUE THREAD

**Issue by cbdbdd** [fchollet/keras#4464]
i'm writing custom elementwise multiplication layers in keras using theano backend with GPU computing. i found that broadcasting rules supported by numpy are not supported by keras. for example:
numpy supports:   numpy.random.random(1,25,26)*numpy.random.random(3,25,26) ->output shape: (3,25,26)
but in keras custom layer, i can't do that in this structure:
input layer:  shape ->(3,25,26)
custom layer:  do elementwise multiplication :   w×input       where w's shape is (1,25,26)

can anyone knows is this a bug or limitation of keras theano backend?as i mentioned in #4454 
many thanks!

Please make sure that the boxes below are checked before you submit your issue. Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by cbdbdd** [comment]
i solved the keras with theano backend elementwise (array broadcasting) layer issue. so the answer is yes,keras support it. solution is just don't specify w's shape with (1,25,26) but (25,26). and i wonder if i can broadcast any axis simply by using k.permute to reorder those axes of the inputs which need to be broadcasted to do the elementwise operations?   not tested yet, but i believe so.

**Comment by keunwoochoi** [comment]
I think this should be documented in somewhere.

**Comment by hanfeisun** [comment]
I used Tensorflow so Theano's backend doesn't work for me.

Is there a general way to do this?
